integrations/ollama/src/ollama_haystack/generator.py

```diff
@@ -0,0 +1,82 @@
+from dataclasses import dataclass
+from datetime import datetime
+from typing import Optional, Any, Dict, List
+from urllib.parse import urljoin
+
+import requests
+from haystack import component
+
+
+@dataclass
+class OllamaResponse:
+    model: str
+    created_at: datetime
+    response: str
+    context: List[int]
+    done: bool
+    total_duration: int
+    load_duration: int
+    prompt_eval_count: int
+    prompt_eval_duration: int
+    eval_count: int
+    eval_duration: int
+
+    def __post_init__(self):
+        self.metadata = {
+            key: value for key, value in self.__dict__.items() if key != "response"
+        }
+
+    def as_haystack_generator_response(self) -> Dict[str, List]:
+        """Returns replies and metadata in the format required by haystack"""
+        return {"replies": [self.response], "metadata": [self.metadata]}
```

Comment:
Lately, we are transitioning to `meta` instead of `metadata`.

Reply:
Will sort that out!